api.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package api;
+
+service ResolverService {
+  rpc Apply(ApplyRequest)
+      returns (ApplyResponse);
+}
+
+message ApplyRequest {
+  SystemResolver resolver = 1;
+}
+
+message ApplyResponse {}
+
+// SystemResolver provides configuration options for the system resolver.
+//
+// https://man7.org/linux/man-pages/man5/resolv.conf.5.html.
```

Comment:
Do we want this to be OS-agnostic?

Reply:
That is a _great_ question. I think @stevendborrelli has opinions here. I think starting with Linux makes sense, but that could make things harder later.

Reply:
I think tackling lowest common denominator from the beginning could be a strategy here. I'm not sure.

Reply:
Sounds like there is enough friction here that we should add it to the agenda for next week?

Reply:
I would also prefer to start with the lowest common denominator from the API perspective. The implementation for every specific OS should know howto materialize it.
But with this in mind, we should start with a small API surface.
resolv.conf is a good example, because i think very few installations require to tweak all  of these  ~20 knobs there.

Given this assumption, we should consider to introduce more meaningful and less OS opinionated property names, for example

`ndots` could potentially be exposed as `NumberOfDots`